core/modules/parsers/parseutils.js

```diff
@@ -313,6 +317,9 @@ exports.parseAttribute = function(source,pos) {
 		node.type = "string";
 		node.value = "true";
 	}
+	if(node.type === "string") {
```

Comment:
This is a non-backwards compatible change that impacts all parsers that use the `parseAttribute` utility.

TW syntax trees require HTML entities to be expressed as a node with type=entity.

Reply:
@Jermolene

There may be some misunderstandings... I thought `parseAttribute()` functions **within** an HTML node. Here's the kind of problem I'm describing:

If you create a regular tiddler with the following text:
```html
<$codeblock code="var s=&quot;It's tricky&quot;;" language="js"/>
```

Without the patch, the rendered output will be this:
```
var s=&quot;It's tricky&quot;;
```

With the patch, you get:
```
var s="It's tricky";
```

Another example:
```html
<span title="this &quot;has&quot; a title">hello world</span>
```

If you hover over the rendered text "hello world", you'll find the title (tooltip) not showing the expected text: `this "has" a title`.

I've traced the problem down to Widget's calling `domNode.setAttributeNS(namespace,name,value)`. If you don't DECODE entities before calling this function, any entity markup with be treated as literal text.

And I believe unescaping entities needs to happen in the parsing stage; otherwise, by the time you get to widget tree stage, it's difficult to discern where a `codeblock` widget's  `code` value comes from. If it comes from `<$codeblock>` text like above, then the value needs decoding; but if it comes from a fenced code block, the content should not be decoded.

Reply:
Hi @Jermolene 

Have you had another chance to look into decoding entities in string literals during ATTRIBUTE parsing? Filtered transclusions and macro values are not impacted by the proposed change.

Reply:
Hi @cdruan this stuff is subtle. It's correct that HTML decodes entities within attribute values, but that only happens within HTML source code. If you assign an attribute in JS using the DOM API then no such decoding takes place. So the following assigns the literal string `one &amp; two` to the attribute:

```
$0.setAttribute("data-my-attrib","one &amp; two")
```

In TiddlyWiki, we often use computed attributes, where we normally wouldn't want the automatic conversion. For example, the wikify widget has an attribute `text` that contains wikitext; it would break things if the HTML entities were decoded before the wikitext were parsed.

In Markdown, of course, perhaps you would want to decode entities in attributes because there is no mechanism like dynamic attributes to complicate things.

Reply:
Thanks @cdruan